ibis/expr/operations.py

```diff
@@ -59,6 +59,18 @@ def _pp(x):
 
         return '{}({})'.format(opname, ', '.join(pprint_args))
 
+    def __getstate__(self):
+        excluded_slots = {'_expr_cached', '_hash'}
+        return {
```

Comment:
I am not certain how it works when slot is another ibis TableExpr or ColumnExpr, can you explain it works?